### ISSUE TITLE
Invalid autocorrection of ternary expression by RedundantDoubleSplatHashBraces

### ISSUE THREAD

**Issue by nevans** [rubocop/rubocop#12286]
rubocop 1.57.1 incorrectly identifies and incorrectly autocorrects the following code:

```ruby
# frozen_string_literal: true

foo(**(bar ? {bar:} : {}))
foo(bar:, **(baz ? {baz:} : {}))
```

My apologies if this was already fixed by #12278, but I don't think it is (and it doesn't match any of the test cases there).  It's related to #12278, at any rate!

--------

## Expected behavior

It should not report an offense, the double splat is not redundant in these examples.  And it especially should not autocorrect to invalid syntax.

## Actual behavior

```console
$ bin/rubocop tmp/rubocop-example.rb -A
Inspecting 1 file
F

Offenses:

tmp/rubocop-example.rb:3:5: C: [Corrected] Style/RedundantDoubleSplatHashBraces: Remove the redundant double splat and b
races, use keyword arguments directly.
foo(**(bar ? {bar:} : {}))
    ^^^^^^^^^^^^^^^^^^^^^
tmp/rubocop-example.rb:3:17: F: Lint/Syntax: unexpected token tCOLON
(Using Ruby 3.1 parser; configure using TargetRubyVersion parameter, under AllCops)
foo((bar ? bar: : {}))
                ^
tmp/rubocop-example.rb:4:11: C: [Corrected] Style/RedundantDoubleSplatHashBraces: Remove the redundant double splat and 
braces, use keyword arguments directly.
foo(bar:, **(baz ? {baz:} : {}))
          ^^^^^^^^^^^^^^^^^^^^^
tmp/rubocop-example.rb:4:23: F: Lint/Syntax: unexpected token tCOLON
(Using Ruby 3.1 parser; configure using TargetRubyVersion parameter, under AllCops)
foo(bar:, (baz ? baz: : {}))
                      ^
tmp/rubocop-example.rb:4:28: F: Lint/Syntax: unexpected token tRPAREN
(Using Ruby 3.1 parser; configure using TargetRubyVersion parameter, under AllCops)
foo(bar:, (baz ? baz: : {}))
                           ^

1 file inspected, 5 offenses detected, 2 offenses corrected
```
After autocorrection:
```ruby
# frozen_string_literal: true

foo((bar ? bar: : {}))
foo(bar:, (baz ? baz: : {}))
```

## Steps to reproduce the problem

See above.

## RuboCop version

```console
$ rubocop -V
1.57.1 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 3.1.4) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.21.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.24.1
```
